src/12-fine-tuning/lib/fine-tune.ts

```diff
@@ -0,0 +1,56 @@
+import { SELECTED_INPUT } from "12-fine-tuning/01-medium"
+import OpenAI from "openai"
+import dotenv from "dotenv"
+
+dotenv.config()
+
+/**
+ * Kicks off a fine-tuning job with the uploaded training and validation datasets.
+ * Continues polling for status until the job is complete.
+ *
+ * @param uploads An object with the references to the uploaded training and validation datasets.
+ * @returns A `OpenAI.FineTuning.Jobs.FineTuningJob` with the final state of the fine-tuning job, whether `succeeded`, `failed`, or `cancelled`.
+ */
+export async function submitFineTuneJob(uploads: {
+  trainingData: OpenAI.Files.FileObject
+  validationData: OpenAI.Files.FileObject
+}) {
+  const {
+    trainingData: { id: training_file },
+    validationData: { id: validation_file },
+  } = uploads
+
+  const openai = new OpenAI()
+  const fineTune = await openai.fineTuning.jobs.create({
+    training_file,
+    validation_file,
+    model: "gpt-3.5-turbo",
+    suffix: SELECTED_INPUT,
+  })
+  console.log({ fineTune })
+
+  let fineTuneState = await openai.fineTuning.jobs.retrieve(fineTune.id)
+
+  const notDone = (status: typeof fineTuneState.status) => {
+    return (
+      status !== "succeeded" && status !== "failed" && status !== "cancelled"
+    )
+  }
+
+  while (notDone(fineTuneState.status)) {
+    fineTuneState = await openai.fineTuning.jobs.retrieve(fineTune.id)
```

Comment:
Takes a few minutes to ~ an hour depending on dataset size. So this loop just polls until it's done.